hangupsbot/commands/__init__.py

```diff
@@ -18,6 +21,182 @@ def __init__(self):
 
         self.command_tagsets = {}
 
+        self.preprocessors = { "inbuilt": { r"(?<!@)@[^@]\w+[^@]$": self.one_chat_id,
+                                            r"(?<!#)#[^#]\w+[^#]$": self.one_conv_id }}
+
+    def one_chat_id(self, token, internal_context):
+        user_memory = self.bot.get_memory_option("user_data")
+        if internal_context:
+            chat_ids = list(self.bot.conversations.catalog[internal_context.conv_id]["participants"])
+        else:
+            chat_ids = list(user_memory.keys())
+
```

Comment:
oops typoed - in the docs, should be "internal_context" i.e. source event and corrected

currently, the source event is always supplied since commands must arise from an actual bot event, so defaults are unnecessary